### DISCUSSION TITLE
`when()` method condition seem to be ignored by the analyzer

### DISCUSSION THREAD

**Discussion by hfoletto** [larastan/larastan#1762] [Q&A]
Hey all, greetings!

I'm starting to use Larastan, so I'm still getting used to how it works.

For this specific question, this is the problem I'm seeing:

I have the following scope defined in my model:
```php
/**
 * @param  Builder<Profile>  $query
 */
public function scopeFilter(Builder $query, ProfileFilterData $filters): void
{
    $query
        ->when(! ($filters->search instanceof Optional), function (Builder $query) use ($filters) {
            $query->where('name', 'like', '%'.$filters->search.'%');
        });
}
```

Larastan returns a problem there, with the following message:
```
Binary operation "." between '%' and Spatie\LaravelData\Optional|string results in an error.  
```

I understand the message, but it's not possible that the type would be `Spatie\LaravelData\Optional` since I'm checking exactly that with the `when` method.

Looks like Larastan is completely ignoring the when condition.

Am I doing something incorrect here?

**Comment by canvural** [comment]
Hi,

Yeah this is a known issue. And currently it is not even possible to fix it. PHPStan does not have an extension point that can influence the type in the closure. So it is a PHPStan limitation.

I vaguely remember there was a feature request about this in PHPStan repo. So please search there. If there is not you can create a feature request there.
